drivers/flash/flash_stm32g4x.c

```diff
@@ -181,18 +205,34 @@ void flash_stm32_page_layout(const struct device *dev,
 			     const struct flash_pages_layout **layout,
 			     size_t *layout_size)
 {
-	static struct flash_pages_layout stm32g4_flash_layout = {
-		.pages_count = 0,
-		.pages_size = 0,
-	};
-
 	ARG_UNUSED(dev);
 
-	if (stm32g4_flash_layout.pages_count == 0) {
-		stm32g4_flash_layout.pages_count = FLASH_SIZE / FLASH_PAGE_SIZE;
-		stm32g4_flash_layout.pages_size = FLASH_PAGE_SIZE;
+#if defined(FLASH_OPTR_DBANK) && (CONFIG_FLASH_SIZE < STM32G4_FAMILY_MAX_FLASH)
+#define PAGES_PER_BANK  ((FLASH_SIZE / FLASH_PAGE_SIZE) / 2)
+	static struct flash_pages_layout stm32g4_flash_layout[3];
+
+	if (stm32g4_flash_layout[0].pages_count == 0) {
+		/* Bank1 */
+		stm32g4_flash_layout[0].pages_count = PAGES_PER_BANK;
+		stm32g4_flash_layout[0].pages_size = FLASH_PAGE_SIZE;
+		/* Dummy page corresponding to discontinuity between bank1/2 */
+		stm32g4_flash_layout[1].pages_count = 1;
+		stm32g4_flash_layout[1].pages_size = BANK2_OFFSET
+					- (PAGES_PER_BANK * FLASH_PAGE_SIZE);
+		/* Bank2 */
+		stm32g4_flash_layout[2].pages_count = PAGES_PER_BANK;
+		stm32g4_flash_layout[2].pages_size = FLASH_PAGE_SIZE;
+	}
+#else
+	static struct flash_pages_layout stm32g4_flash_layout[1];
+
+	if (stm32g4_flash_layout[0].pages_count == 0) {
+		stm32g4_flash_layout[0].pages_count = FLASH_SIZE
+						/ FLASH_PAGE_SIZE;
+		stm32g4_flash_layout[0].pages_size = FLASH_PAGE_SIZE;
 	}
+#endif
 
-	*layout = &stm32g4_flash_layout;
-	*layout_size = 1;
+	*layout = stm32g4_flash_layout;
```

Comment:
The `&` has to be kept here because `layout` is a pointer to a pointer.
```suggestion
	*layout = &stm32g4_flash_layout;
```

Reply:
Oops, forget about this. It's too late... we have an array now, so this is totally correct. Otherwise would have been strange also that it worked in Azeemsha's test.